src/gui/qgsabstractdatasourcewidget.h

```diff
@@ -60,6 +54,21 @@ class GUI_EXPORT QgsAbstractDataSourceWidget : public QDialog
      */
     void setBrowserModel( QgsBrowserModel *model );
 
+    /**
+     * Returns the dialog map canvas
+     * \see setMapCanvas()
+     *
+     * \since QGIS 3.26
+     */
+    virtual QgsMapCanvas *mapCanvas() = 0;
+
+    /**
+     * Sets the dialog map canvas
+     * \see mapCanvas()
+     * \since QGIS 3.26
+     */
+    virtual void setMapCanvas( QgsMapCanvas *mapCanvas ) = 0;
```

Comment:
Again, no need to make this pure virtual -- we make it virtual and leave the base class handling untouched (ie it just sets mMapCanvas, like it currently does).

```suggestion
    virtual void setMapCanvas( QgsMapCanvas *mapCanvas )
```